src/ecmult_impl.h

```diff
@@ -220,9 +220,24 @@ static int secp256k1_ecmult_wnaf(int *wnaf, int len, const secp256k1_scalar *a,
     return last_set_bit + 1;
 }
 
+/* Same as secp256k1_ecmult_wnaf, but stores to int8_t array. Requires w <= 8. */
+static int secp256k1_ecmult_wnaf_small(int8_t *wnaf, int len, const secp256k1_scalar *a, int w) {
+    int wnaf_tmp[256];
+    int ret, i;
+
+    VERIFY_CHECK(2 <= w && w <= 8);
+    ret = secp256k1_ecmult_wnaf(wnaf_tmp, len, a, w);
+
+    for (i = 0; i < len; i++) {
+        wnaf[i] = (int8_t)wnaf_tmp[i];
+    }
+
+    return ret;
+}
+
 struct secp256k1_strauss_point_state {
-    int wnaf_na_1[129];
-    int wnaf_na_lam[129];
+    int8_t wnaf_na_1[129];
```

Comment:
```suggestion
    int_least8_t wnaf_na_1[129];
```

This is technically better to retain support for platforms that don't offer `int8_t`. I don't think there are any relevant platforms of that kind, but why not... 🤷  

edit: Now that I think about it again, a platform that does not offer `int8_t` will most likely one for which `CHAR_BITS != 8`. And then it's neither relevant nor it's likely to offer `int32_t` or `int64_t` which we need anyway. So I guess there's no difference between `int8_t` and `int_least8_t`.